src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -56,4 +57,14 @@ public MovieSession add(MovieSession movieSession) {
             return query.getResultList();
         }
     }
+
+    @Override
+    public Optional<MovieSession> findById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            return Optional.of(session.get(MovieSession.class, id));
```

Comment:
again, what happens if we call Optional.of(null)?

Reply:
Will be NPE. Sorry, i'm blind